project_task_required_project/models/project_task.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2025 Ángel García de la Chica Herrera <angel.garcia@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+
+    @api.constrains("project_id")
+    def _check_project_id_not_null(self):
+        if self.env.company.is_project_required and any(
```

Comment:
Shouldn't you check the project's company instead?

Reply:
I think not, since what you want to check is if the project is not null.

Reply:
Yes, but the fact that setting a project is mandatory or not depends on the company's configuration, so I think that the company set in the task should determine if the project must be set or not. You are using the env here, which I don't think is correct.

Reply:
I think you are right. Done!